docs/src/Theory/Atmos/Microphysics_0M.md

```diff
@@ -0,0 +1,92 @@
+# Microphysics_0M
+
+The `Microphysics_0M.jl` module defines a 0-moment bulk parameterization of
+  precipitation removal.
+It offers a very simple way of removing the excess water from the system without
+  assuming anything about the size distributions of cloud
+  or precipitation particles.
+
+The `q_tot` (total water specific humidity) tendency
+  is obtained by relaxation with a constant timescale
+  to a state with precipitable water removed.
+The threshold for when to remove `q_tot` is defined either by the
+  condensate specific humidity or supersaturation.
+The thresholds and the relaxation timescale are defined in
+  CLIMAParameters.
+
+!!! note
+
+    The implementation assumes that
+    `q_liq` (liquid water specific humidity) and
+    `q_ice` (ice water specific humidity) are in the auxiliary variables.
+
+!!! note
+
+    To remove precipitation instantly the relaxation timescale should be
+    equal to the timestep length.
+    Not sure if we want to implement it this way though.
+
+## Precipitation removal tendency
+
+If based on maximum condensate specific humidity the tendency is defined as:
+``` math
+\begin{equation}
+  \left. \frac{d \, q_{tot}}{dt} \right|_{precip} =-
+    \frac{max(0, q_{liq} + q_{ice} - q_{c0})}{\tau_{precip}}
+\end{equation}
+```
+where:
+  - `q_{liq}`, `q_{ice}` are cloud liquid water and cloud ice specific humidities,
+  - `q_{c0}` is the condensate specific humidity threshold above which water is removed,
+  - `\tau_{precip}` is the relaxation timescale.
+
+If based on saturation excess the tendency is defined as:
+```math
+\begin{equation}
+  \left. \frac{d \, q_{tot}}{dt} \right|_{precip} =-
+    \frac{max(0, q_{liq} + q_{ice} - S_{0} \, q_{vap}^{sat})}{\tau_{precip}}
+\end{equation}
+```
+where:
+  - `q_{liq}`, `q_{ice}` are cloud liquid water and cloud ice specific humidities,
+  - `S_{0}` is the supersaturation threshold above which water is removed,
+  - `q_{vap}^{sat}` is the saturation specific humidity,
+  - `\tau_{precip}` is the relaxation timescale.
+
+## Coupling to the state variables
+
+Removing the `q_tot` changes the mass of the working fluid.
+Therefore the state variables `\rho` and `\rho q_{tot}` are by:
+
+```math
```

Comment:
Better to express this as the source in the conservation law, which is not the same as the material derivative (but is what we need here). See Eq. (3.7) in the design doc.

Reply:
I have updated the docs and the code to follow the notation from the DDs. I'm still not sure about the total energy source term. I will send the link to the documentation once it is generated.

Reply:
https://clima.github.io/ClimateMachine.jl/previews/PR1440/Theory/Atmos/Microphysics_0M/

Reply:
I looked at the docs. I'm still confused by the c_v,d term. For example when you remove q_liq (assuming no_ice) you lose the total energy of liquid. It looks strange to me that the source term is zero when c_v,l = c_v,d?

Reply:
ok - I think this is wrong and you are right. Will update again!